components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/prometheus/reporter/PrometheusMetricCreatorUtils.java

```diff
@@ -0,0 +1,143 @@
+                "Total number of requests to an api", map);
+        prometheusReporter.createMetric(SynapseConstants.FAIL_SAFE_MODE_API, MetricConstants.HISTOGRAM,
+                MetricConstants.API_LATENCY_SECONDS,
+                "Latency of requests to an api", map);
+    }
+
+    /**
+     * Create the inbound endpoint related metrics.
+     */
+    public static void createInboundEndpointMetric() {
+        HashMap<String, String[]> map = new HashMap<>();
+        String[] labels = {MetricConstants.SERVICE_NAME, MetricConstants.SERVICE_TYPE};
+        map.put(MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_TOTAL, labels);
+        map.put(MetricConstants.INBOUND_ENDPOINT_LATENCY_SECONDS, labels);
+
+        prometheusReporter.createMetric("INBOUND_ENDPOINT", MetricConstants.COUNTER,
+                MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_TOTAL,
+                "Total number of requests to an inbound endpoint.", map);
+        prometheusReporter.createMetric("INBOUND_ENDPOINT", MetricConstants.HISTOGRAM,
+                MetricConstants.INBOUND_ENDPOINT_LATENCY_SECONDS,
+                "Latency of requests to an inbound endpoint.", map);
+    }
+
+    /**
+     * Create the metrics related to failed proxy services.
+     */
+    public static void createProxyServiceErrorMetric() {
+        HashMap<String, String[]> map = new HashMap();
+        map.put(MetricConstants.PROXY_REQUEST_COUNT_ERROR_TOTAL, new String[]{MetricConstants.SERVICE_NAME,
+                MetricConstants.SERVICE_TYPE});
+        new PrometheusReporter().initErrorMetrics("PROXY", MetricConstants.COUNTER,
+                MetricConstants.PROXY_REQUEST_COUNT_ERROR_TOTAL,
+                "Total number of error requests to a proxy service", map);
+    }
+
+    /**
+     * Create the metrics related to failed apis.
+     */
+    public static void createApiErrorMetric() {
+        HashMap<String, String[]> map = new HashMap();
+
+        map.put(MetricConstants.API_REQUEST_COUNT_ERROR_TOTAL, new String[]{MetricConstants.SERVICE_NAME,
+                MetricConstants.SERVICE_TYPE, MetricConstants.INVOCATION_URL});
+        prometheusReporter.initErrorMetrics("API", MetricConstants.COUNTER,
+                MetricConstants.API_REQUEST_COUNT_ERROR_TOTAL,
+                "Total number of error requests to an api", map);
+    }
+
+    /**
+     * Create the metrics related to failed inbound endpoints.
+     */
+    public static void createInboundEndpointErrorMetric() {
+        HashMap<String, String[]> map = new HashMap();
+
+        map.put(MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_ERROR_TOTAL, new String[]{MetricConstants.SERVICE_NAME,
+                MetricConstants.SERVICE_TYPE});
+        prometheusReporter.initErrorMetrics("INBOUND_ENDPOINT", MetricConstants.COUNTER,
+                MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_ERROR_TOTAL,
+                "Total number of error requests to an inbound endpoint.", map);
```

Comment:
Does this metric report errors during mediating a request taken into an inbound EP? Or is it reporting errors which occurred during receiving the message by inbound EP? I do not get what is meant by `error requests`.

Reply:
Fixed